lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -566,8 +566,8 @@ def self.config_type
     tf = Types::TypeFactory
     nes_t = Types::PStringType::NON_EMPTY
 
-    # Need alias here to avoid ridiculously long regexp burp in case of validation errors.
-    uri_t = Pcore::TYPE_URI_ALIAS
+    # Validated using Ruby URI implementation
```

Comment:
Where does that take place? What does the error message look like? Add a test that shows that?

Reply:
There is a test (checks for the text "bad URI"). It's generated by the Ruby URI constructor and looks like this:
```
URI::InvalidURIError: bad URI(is not URI?): dothis -f bar
```

Reply:
I added a test with "dothis%20-f%20bar"